api/fbs/v1/schema.fbs

```diff
@@ -0,0 +1,121 @@
+    // duration determines how long a TraceSeries keeps its data
+    // unit    = "ns" | "ms" | "s" | "m" | "h" | "d" | "w" .
+    // duration: int_literal unit
+    duration:string;
+    // updated_at indicates when the TraceSeries is updated
+    updated_at:uint64;
+}
+
+// Catalog refers to a placement contains objects belonged to a particular data type
+enum Catalog:byte { Trace = 0, Log = 1, Metric = 2 }
+
+// IndexType determine the index structure under the hood
+enum IndexType:byte { Text = 0, Numerical = 1, ID = 2}
+
+// IndexObject defines who should be indexed.
+table IndexObject {
+    // fields are the combination that refers to an indexed object
+    // If the elements in fields are more than 1, the object will generate a multi-field index
+    // Caveat: All fields in a multi-field index MUST have an identical IndexTypee
+    fields:[string];
+    // type is the IndexType of this IndexObject.
+    type:IndexType;
+}
+
+// IndexRule defines how to generate indices based on IndexObject
+// IndexRule should bind to an IndexSubject through an IndexRuleBinding to generate proper indices.
+// Example: A trace entity wants to index fields: trace_id, service_name, endpoint_name, and latency.
+//          and service_name and endpoint_name would combine a multi-field index.
+// The index rule could be:
+// IndexRule {
+//     metadata: {
+//         name: sw_trace
+//         group: production
+//     }
+//     objects: [
+//        { 
+//            fields: [trace_id]
+//            type: ID
+//        },
+//        { 
+//            fields: [service_name, endpoint_name]
+//            type: Text
+//        },
+//        { 
+//            fields: latency]
+//            type: Numerical
+//        },
+//    ]
+//    update_at: .......
+// } 
+table IndexRule {
+    // metadata define the rule's identity
+    metadata:Metadata;
+    // objects refer to which fields should be indexed
+    objects:[IndexObject];
+    // updated_at indicates when the IndexRule is updated
+    updated_at:uint64;
+}
+
+// IndexSubject defines which subject(series) would generate indices
+// For example, if a TraceSeries's metadata is {name: sw_trace, group: production},
+//                in consequence, the IndexSubject is
+//                Index Subject {
+//                    catalog: Trace
+//                    series: {name: sw_trace, group: production}
+//                }
+table IndexSubject {
+    // catalog is where the subject/series belongs to
+    catalog:Catalog;
+    // series refers to a series in a particular catalog
+    series:Metadata;
+}
+
+
+// IndexRuleBinding is a bridge to connect an IndexRule to several IndexSubjects
+// This binding is valid between begin_at and expire_at, that provides flexible strategies
+// to control how to generate time series indices.
+table IndexRuleBinding {
+    // metadata is the identity of this binding
+    metadata:Metadata;
+    // rule_ref refers to the IndexRule
+    rule_ref:Metadata;
```

Comment:
typo?

Reply:
No. It's a reference to IndexRule by its metadata(name and group)

Reply:
I see